tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -2248,17 +2302,18 @@ static void resolveCursorFromUSR(
 }
 
 void SwiftLangSupport::getCursorInfoFromUSR(
-    StringRef filename, StringRef USR, bool CancelOnSubsequentRequest,
-    ArrayRef<const char *> Args, Optional<VFSOptions> vfsOptions,
+    StringRef PrimaryFile, StringRef InputFile, StringRef USR,
+    bool CancelOnSubsequentRequest, ArrayRef<const char *> Args,
+    Optional<VFSOptions> vfsOptions,
     SourceKitCancellationToken CancellationToken,
     std::function<void(const RequestResult<CursorInfoData> &)> Receiver) {
   std::string error;
 
-  auto fileSystem = getFileSystem(vfsOptions, filename, error);
+  auto fileSystem = getFileSystem(vfsOptions, InputFile, error);
```

Comment:
Why do you use `InputFile` here but `PrimaryFile` almost everywhere else? I’m not saying this is wrong, I just don’t understand what drove the decision for `InputFile` vs `PrimaryFile` yet.

Reply:
They should all be `InputFile`. I only see the one `PrimaryFile` in diagnostics. That one is a little confusing because diagnostics don't take an input file, only a primary file. Things are even more confusing because `getFileSystem` used to take a `primaryFile` and I've now changed what that actually means - we previously used "source file", "primary file", and "input file" interchangeably. I'll fix up all the callsites that currently have `primaryFile=` as their comment (since I renamed the arg).

`InputFile` == buffer we're doing the request on, could be a generated buffer name
`PrimaryFile` == file to build AST for
`SourceFile` == I'd prefer this isn't used, it can really mean either

Reply:
I think I've changed my mind here. It doesn't really make sense to be able to provide a VFS for a generated input buffer name. So I'll change this back to use primary file.

Reply:
OK, makes sense to me 👍🏽